src/app/src/main/java/com/cmput301f21t09/budgetprojectname/ViewHabitActivity.java

```diff
@@ -27,75 +78,125 @@ protected void onCreate(Bundle savedInstanceState) {
-            habitEventDataList.add(new HabitEventModel());
-        }
+        // Retrieve the specific view
+        habitEventList = findViewById(R.id.past_habit_event_list);
 
+        // Set up the list and send an empty list to the view
+        habitEventDataList = new ArrayList<>();
         habitEventAdapter = new HabitEventCustomList(this, habitEventDataList);
         habitEventList.setAdapter(habitEventAdapter);
 
+        // Fetches the past habit events related to the current habit from Firestore using habitID
+        // and update the view
+        habitEventController.readHabitEvent(habitID, new HabitEventController.HabitEventListCallback() {
+            @Override
+            public void onCallback(ArrayList<HabitEventModel> hbEvtLst) {
+                habitEventDataList.clear();
+                habitEventDataList.addAll(hbEvtLst);
+                habitEventAdapter.notifyDataSetChanged();
+            }
+        });
+
         // Go Back
         ImageButton back = findViewById(R.id.view_habit_back_button);
         back.setOnClickListener(v -> {
             finish();
         });
 
+        // Edit habit
+        // Todo: Implement edit Habit function
+        final Button editHabitBtn = findViewById(R.id.editHabitButton);
+        editHabitBtn.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+            }
+        });
+
         // Delete Habit
         ImageButton remove = findViewById(R.id.view_habit_remove_button);
         remove.setOnClickListener(v -> {
             // TODO: Remove Targeted Habit. Dialog To Confirm?
         });
     }
+
+    /**
+     * Update the views of the habit details
+     */
+    private void updateView() {
+        /**
+         * Get the image views
+         */
+        sundayIcon = findViewById(R.id.sunday_icon);
+        mondayIcon = findViewById(R.id.monday_icon);
+        tuesdayIcon = findViewById(R.id.tuesday_icon);
+        wednesdayIcon = findViewById(R.id.wednesday_icon);
+        thursdayIcon = findViewById(R.id.thursday_icon);
+        fridayIcon = findViewById(R.id.friday_icon);
+        saturdayIcon = findViewById(R.id.saturday_icon);
+
+        // Set the views of the habit details accordingly
+        if(controller.isTaskComplete(HabitController.HABIT_MODEL_LOAD)){
+            // Set the toolbar title, habit tile, habit reason, and habit date view
+            IHabitModel model = controller.getModel();
+            habitTitleToolbar.setText(model.getTitle());
+            habitTitle.setText(model.getTitle());
+            habitDescription.setText(model.getReason());
+            habitDate.setText(new SimpleDateFormat("MM/dd/yyyy").format(model.getStartDate()));
+
+            // Set the frequency of habit accordingly
+            IWeeklyHabitScheduleModel schedule = (IWeeklyHabitScheduleModel) model.getSchedule();
+            if (schedule.getDay(IWeeklyHabitScheduleModel.SUNDAY))
```

Comment:
I set the negative icon as a default for everyday in the layout so I think it would be better if we just set positive programmatically

Reply:
Fixed!

Reply:
I'll approve for note that I have created a fragment for displaying the schedule you can use now!

Reply:
Sounds good! I'll get refactor it and create a separate PR for that. :)